### ISSUE TITLE
Reconsider Potential Merge Conflicts file decorations

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/vscode-remote-repositories-github#123]
RemoteHub displays file decorations for files containing uncommitted changes which also have incoming changes based on the auto fetch that RemoteHub periodically runs. @Tyriar had the following feedback:

> That icon generally means something is wrong though and it's not yet. I think you said it's because changes have not yet been pulled, the git way to me is that it wouldn't complain until the merge is attempted and it fails as the typical case imo is that there aren't merge conflicts

I personally find these file decorations valuable, albeit because the potential conflict decorations provide a way to notify the user that their file contents may be out of date. Perhaps we can consider a setting to disable the potential conflict decorations.

**Comment by Tyriar** [comment]
If a notification that the _repo_ is out of date should the warning on the repo not the file? Would some icon indicating it's stale (maybe a problem) be better than a warning (there is a problem)? Can we query GH whether it can be merged instead and show a warning only when there's a problem, ie. the `Update Branch` button is disabled?

> Perhaps we can consider a setting to disable the potential conflict decorations.

I doubt it's worth adding a setting, I'd expect the amount of usage it would get would be very small.

**Comment by joyceerhl** [comment]
> If a notification that the repo is out of date should the warning on the repo not the file? 

We do also already have the status bar item in the bottom left which indicates whether there are incoming changes for a file, but that doesn't tell me whether specific files I'm working on are outdated and might benefit from a sync.

> Would some icon indicating it's stale (maybe a problem) be better than a warning (there is a problem)?

Yeah I think that is the intention of these file decorations. Perhaps we do need a new icon? cc @bamurtaugh @misolori

**Comment by bamurtaugh** [comment]
> Perhaps we do need a new icon?

Including an icon to indicate there's a potential problem sounds good to me. To me, the yellow triangle would indicate there might be a problem and the user should investigate. @misolori do you have any idea for another icon, or how to make this icon potentially even better fit the scenario (i.e. should we show it top-level at the repo-level like @Tyriar suggested)?

**Comment by misolori** [comment]
@joyceerhl I believe we are limited in the decorators by using unicode characters, is that still the case? Otherwise the `cloud-download` icon could work:

<img width="142" alt="CleanShot 2022-02-15 at 21 46 27@2x" src="https://user-images.githubusercontent.com/35271042/154203970-dda7857e-d11d-4c42-aab1-41b08f728df8.png">

<img width="274" alt="CleanShot 2022-02-15 at 21 48 23@2x" src="https://user-images.githubusercontent.com/35271042/154204172-60e391aa-e87d-4622-8d78-2c62f70000cb.png">

**Comment by joyceerhl** [comment]
Yeah I just tried and it looks like file decorations don't support the codicons: https://github.com/microsoft/vscode/issues/135591

**Comment by joyceerhl** [comment]
I think we now have the ability to use codicons in decorations via proposed API. The work left here is
1. Pick a codicon
2. Apply the codicon in file decorations (note these will only appear in the editor title area and not in the SCM viewlet since those files are technically not tracked changes yet)
